### PR TITLE
Move to Somfy OAuth as access method for Somfy Europe

### PR DESCRIPTION
**Pull request by iMicknl**
Not pretty code, however we need to see if this will fix the issues.

### CODE REVIEW

File: pyoverkiz/client.py [hunk 1]
```diff
@@ -155,6 +173,78 @@ async def login(
...
+            token = await response.json()
+            # { "message": "error.invalid.grant", "data": [], "uid": "xxx" }
+            if "message" in token and token["message"] == "error.invalid.grant":
+                raise SomfyBadCredentialsException(token["message"])
+
+            if "access_token" not in token:
+                raise SomfyServiceException("No Somfy access token provided.")
+
+            self._access_token = str(token["access_token"])
+            self._refresh_token = str(token["refresh_token"])
```
Comment (tetienne) [inline pyoverkiz/client.py:241]: They are not already a string?

File: pyoverkiz/client.py [hunk 2]
```diff
@@ -155,6 +173,78 @@ async def login(
...
+            self._expires_in = datetime.datetime.now() + datetime.timedelta(
+                seconds=int(token["expires_in"]) - 5
```
Comment (tetienne) [inline pyoverkiz/client.py:243]: It’s not already an int into the response?

File: pyoverkiz/client.py [hunk 3]
```diff
@@ -508,25 +601,45 @@ async def execute_scenario(self, oid: str) -> str:
...
             return await response.json()
 
     async def __delete(self, path: str) -> None:
         """Make a DELETE request to the OverKiz API"""
-        async with self.session.delete(f"{self.server.endpoint}{path}") as response:
+        headers = {}
+
+        if self.server == SUPPORTED_SERVERS["somfy_europe"]:
+            await self._refresh_somfy_tahoma_token_if_expired()
+            headers["Authorization"] = f"Bearer {self._access_token}"
```
Comment (tetienne) [inline pyoverkiz/client.py:638]: Can put somewhere else later to avoid dup code.

**Review by tetienne — APPROVED** [review]
Just think about removing all the print statements, and for a 1st version,it’s ok. 

We will be able to clean it later.

File: pyoverkiz/client.py (hunk 3 above)
Comment (iMicknl) [reply pyoverkiz/client.py:638]: I thought about that one, and pass the headers to that function, however for now I found this clear enough.

Would this be ok for this current fix?

File: pyoverkiz/client.py (hunk 1 above)
Comment (iMicknl) [reply pyoverkiz/client.py:241]: I think so... Would need to double check. Same for the int.

**Review by tetienne — APPROVED** [review]

**Pull request merged**